### ISSUE TITLE
Table closed also removes

### ISSUE THREAD

**Issue by koertho** [contao/core-bundle#1035]
Setting 'closed' to true in dca ($GLOBALS['TL_DCA'][$strTable]['list']['configuration'] also removes the backlink, if 'global_operations' is empty or not set.

Contao 4.4.3
PHP 7.1.6
Apache 2.4.18
Server: Ubuntu 16.04 LTS

**Comment by leofeyer** [comment]
What do you mean by "backlink"?

**Comment by koertho** [comment]
This one:
![backlink](https://user-images.githubusercontent.com/12064642/29568572-07ed4dc4-8751-11e7-9255-d5de117c6797.png)

I've create a backend module which is called from the news backend module as global operation. From this module I want to go back. I have no global operations there and don't want the option to add new entries (The "Gesendete Newsalert" was just added to see the backlink).

**Comment by leofeyer** [comment]
Can you post your DCA configuration so I can reproduce the case please?

**Comment by koertho** [comment]
```
$GLOBALS['TL_DCA'][$strTable] = [
    'config'   => [
        'dataContainer'    => 'Table',
        'switchToEdit'     => false,
        'enableVersioning' => false,
        'closed'           => true,
        'backlink'         => 'do=news',
        'label'            => $translator->trans('hh.newsalert.tl_newsalert_sent.label'),
        'sql'              => [
            'keys' => [
                'id' => 'primary',
            ]
        ]
    ],
    'list'     => [
        'sorting'           => [
            'mode'         => 2,
            'fields'       => ['senddate DESC'],
            'flag'         => 8,
            'panelLayout'  => 'debug;filter;sort,search,limit',
        ],
        'label'             => [
            'fields'      => ['senddate', 'pid:tl_news.headline', 'topics', 'user:tl_user.name', 'count_messages'],
            'showColumns' => true,
        ],
        'operations'        => [
            'show'   => [
                'label'     => [
                    $translator->trans('hh.newsalert.operations.show.0'),
                    $translator->trans('hh.newsalert.operations.show.1')
                ],
                'href'  => 'act=show',
                'icon'  => 'show.gif'
            ]
        ]
    ],
...
```
When I set the closed to true, the backlink disappears (it returns when set to false or when I add an global operation).

**Comment by leofeyer** [comment]
I cannot reproduce the issue. In addition, it seems logical to me that there is no "go back" link in DCA sorting mode 2, because there is nothing to go back to (e.g. see `tl_user.php`).

Is this problem reproducible in the core bundle?

**Comment by koertho** [comment]
Yes, I can reproduce is in the news feed module. Set `$GLOBALS['TL_DCA']['tl_news_feed']['config']['closed'] = true` and remove the global operations, then the backlink will disappear.

**Comment by patrickjDE** [comment]
This is caused by https://github.com/contao/core-bundle/blob/master/src/Resources/contao/drivers/DC_Table.php#L4687

Simply adding ` || isset($GLOBALS['TL_DCA'][$this->strTable]['config']['backlink'])` to the if condition would fix the issue, but to cover all use cases the conditions for the default, not explicitly defined backlinks (` || \Input::get('act') == 'select' || $this->ptable`) should probably also be added.

Also, in my opinion the Messages should be moved out of the if clause, I don't see any connection between them and the buttons. (But as it seems unrelated, that might be better suited for another issue)

**Comment by leofeyer** [comment]
@patrickjDE Your fix and your remark regarding the messages are correct. However, I don't understand 

> to cover all use cases the conditions for the default, not explicitly defined backlinks (` || \Input::get('act') == 'select' || $this->ptable`) should probably also be added

Which cases are not covered yet?

**Comment by patrickjDE** [comment]
There are 3 cases that add a backlink in that first giant ternary statement (Lines [4695](https://github.com/contao/core-bundle/blob/master/src/Resources/contao/drivers/DC_Table.php#L4695) - 4697)
 * `$GLOBALS['TL_DCA'][$this->strTable]['config']['backlink']` is set
 * `\Input::get('act') == 'select'`
 * `$this->ptable` evaluates to true

They all have to be added to the surrounding if statement to be rendered on a closed table without global operations.

**Comment by leofeyer** [comment]
But if there are no global operations, `\Input::get('act') == 'select'` can never be true, because "edit multiple" is a global operation itself. 😉

**Comment by leofeyer** [comment]
Fixed in f6699875bb3156fbc2bb673637fe972cf2fd5f9a.
